### PR TITLE
JBIDE-24492 Custom template info is loaded after path changed in

### PR DESCRIPTION
**Pull request by jkopriva**
ApplicationWizard

Signed-off-by: Josef Kopriva <jkopriva@redhat.com>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/ApplicationSourceListPage.java [hunk 1]
```diff
@@ -327,7 +335,28 @@ private IObservableValue createLocalTemplateControls(TabFolder tabContainer,
 				btnBrowseFiles, btnBrowseWorkspaceFiles);
 		return localTemplateFilename;
 	}
-
+	
+	private void refreshWizardAfterLocationChange(String filename) {
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/ApplicationSourceListPage.java]: I would have chosen a better method name which reflects more what this is doing. Ex. ```loadApplicationSource```

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/ApplicationSourceListPage.java [hunk 2]
```diff
@@ -327,7 +335,28 @@ private IObservableValue createLocalTemplateControls(TabFolder tabContainer,
...
+		if (model.isUseLocalAppSource()) {
+			if (StringUtils.isNotBlank(filename) && (OpenshiftUIConstants.URL_VALIDATOR.isValid(filename) || isFile(filename))) {
+				Job job = new Job(NLS.bind("Loading application source", filename)) {
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/ApplicationSourceListPage.java]: maybe this job could be unified with the identical one that's at l809

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/ApplicationSourceListPage.java [hunk 3]
```diff
@@ -308,6 +308,14 @@ private IObservableValue createLocalTemplateControls(TabFolder tabContainer,
 				.validatingBeforeSet(validator).to(BeanProperties
 						.value(IApplicationSourceListPageModel.PROPERTY_LOCAL_APP_SOURCE_FILENAME).observe(model))
 				.in(dbc);
+		
+		DataBindingUtils.addDisposableValueChangeListener(new IValueChangeListener() {
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/ApplicationSourceListPage.java]: works well and is expressive. Alternatively one could have written this as lambda:
```
		DataBindingUtils.addDisposableValueChangeListener((ValueChangeEvent event) -> {
			refreshWizardAfterLocationChange(localTemplateFilename.getValue().toString());
			asyncRedraw(parent);
		}
```

**Comment by jkopriva** [comment]
@adietish Thank you for review! Good catches and very appreciated. I have updated code, can you please review once again?

**Comment by jkopriva** [comment]
@adietish Thx for review and merge!

**Pull request merged**